src/test/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundleLocalHostPortTest.java

```diff
@@ -0,0 +1,279 @@
+    private final MetricRegistry metricRegistry = mock(MetricRegistry.class);
+    private final LifecycleEnvironment lifecycleEnvironment = new LifecycleEnvironment(metricRegistry);
+    private final Environment environment = mock(Environment.class);
+    private final Bootstrap<?> bootstrap = mock(Bootstrap.class);
+    private final Configuration configuration = mock(Configuration.class);
+
+    static {
+        val root = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+        root.setLevel(Level.INFO);
+    }
+
+
+    private final ServiceDiscoveryBundle<Configuration> bundle = new ServiceDiscoveryBundle<Configuration>() {
+        @Override
+        protected ServiceDiscoveryConfiguration getRangerConfiguration(Configuration configuration) {
+            return serviceDiscoveryConfiguration;
+        }
+
+        @Override
+        protected String getServiceName(Configuration configuration) {
+            return "TestService";
+        }
+
+
+    };
+
+    @AfterEach
+    public void afterMethod() {
+        DnsCacheManipulator.clearDnsCache();
+    }
+
+    private ServiceDiscoveryConfiguration serviceDiscoveryConfiguration;
+
+
+    @Test
+    void shouldNotAllowPublishingLocalHostAddressToRemoteZk() {
+        DnsCacheManipulator.setDnsCache("myzookeeper", "19.10.1.1");
+        DnsCacheManipulator.setDnsCache("myfavzookeeper", "127.0.0.1");
+        DnsCacheManipulator.setDnsCache("custom-host", "127.0.0.1");
+
+        IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class, () -> {
+            when(jerseyEnvironment.getResourceConfig()).thenReturn(new DropwizardResourceConfig());
+            when(environment.jersey()).thenReturn(jerseyEnvironment);
+            when(environment.lifecycle()).thenReturn(lifecycleEnvironment);
+            when(environment.healthChecks()).thenReturn(healthChecks);
+            when(environment.getObjectMapper()).thenReturn(new ObjectMapper());
+            AdminEnvironment adminEnvironment = mock(AdminEnvironment.class);
+            doNothing().when(adminEnvironment)
+                    .addTask(any());
+            when(environment.admin()).thenReturn(adminEnvironment);
+
+            serviceDiscoveryConfiguration = ServiceDiscoveryConfiguration.builder()
+                    .zookeeper("myzookeeper:2181,myfavzookeeper:2181")
+                    .namespace("test")
+                    .environment("testing")
+                    .connectionRetryIntervalMillis(5000)
+                    .publishedHost("custom-host")
+                    .publishedPort(8021)
+                    .initialRotationStatus(true)
+                    .build();
+            bundle.initialize(bootstrap);
+            bundle.run(configuration, environment);
+
+        });
+
+        assertTrue(thrown.getMessage()
+                .contains("Not allowed to publish localhost address to remote zookeeper"));
+
+    }
+
+    @Test
+    void shouldThrowExceptionWhenUsingUnresolvableZkHost() {
```

Comment:
Rename these methods to smaller names > Junit 5 supports `@DisplayName`, please use that

Reply:
renamed methods